### PR TITLE
Refactor hyperbolic.py with _plotter.py

### PR DESCRIPTION
**Pull request by alicemo1**
Changes made to hyperbolic.py file for homework 1.

## Checklist

- [ ] My pull request has a clear and explanatory title.
- [ ] If necessary, my code is [vectorized](https://www.geeksforgeeks.org/vectorization-in-python/).
- [ ] I added appropriate unit tests.
- [ ] I made sure the code passes all unit tests. (refer to comment below)
- [ ] My PR follows [PEP8](https://peps.python.org/pep-0008/) guidelines. (refer to comment below)
- [ ] My PR follows [geomstats coding style](https://github.com/geomstats/geomstats/blob/master/docs/contributing.rst#coding-style-guidelines) and API.
- [ ] My code is properly documented and I made sure the documentation renders properly. ([Link](https://github.com/geomstats/geomstats/blob/master/docs/contributing.rst#documentation))
- [ ] I linked to issues and PRs that are relevant to this PR.

### CODE REVIEW

**Review by ninamiolane — CHANGES_REQUESTED** [review]
@luisfpereira

File: geomstats/visualization/hyperbolic.py [hunk 1]
```diff
@@ -1,44 +1,33 @@
...
+    """Class used to plot KleinDisk"""
 
-class KleinDisk:
     def __init__(self, points=None):
+        super().__init__()
+
         self.center = gs.array([0.0, 0.0])
         self.points = []
         if points is not None:
             self.add_points(points)
```
Comment (luisfpereira) [inline geomstats/visualization/hyperbolic.py:19]: one of the main ideas of `Plotter` is to not have state, i.e. `self.points` is to be removed. Therefore, this should be removed.

File: geomstats/visualization/hyperbolic.py [hunk 2]
```diff
@@ -51,64 +40,42 @@ def convert_to_klein_coordinates(points):
...
-        circle = plt.Circle((0, 0), radius=1.0, color="black", fill=False)
+    def plot_space(self, ax=None, color="black", **kwargs):
+        """Plot points in the KleinDisk"""
+        ax = self.set_ax(ax=ax)
+
+        circle = plt.Circle((0, 0), radius=1.0, color=color, fill=False)
         ax.add_artist(circle)
         points_x = [gs.to_numpy(point[0]) for point in self.points]
         points_y = [gs.to_numpy(point[1]) for point in self.points]
         ax.scatter(points_x, points_y, **kwargs)
```
Comment (luisfpereira) [inline geomstats/visualization/hyperbolic.py:51]: `plot_space` should be responsible to plot only "the space", without points. Here usually we mean some wireframe style plot.

`plot` receives `points` (check in `Plotter`) and is where points can actually be plotted.

File: geomstats/visualization/hyperbolic.py [hunk 3]
```diff
@@ -51,64 +40,42 @@ def convert_to_klein_coordinates(points):
...
+        self._ax_scale = 1.2
+        self._dim = 2
+        self._convert_points = self._convert_to_poincare_coordinates
+
+    def _convert_to_poincare_coordinates(points):
+        """Convert points to poincare coordinates"""
         poincare_coords = points[:, 1:] / (1 + points[:, :1])
         return poincare_coords
 
     def draw(self, ax, **kwargs):
```
Comment (luisfpereira) [inline geomstats/visualization/hyperbolic.py:77]: this does not exist in `Plotter`.

File: geomstats/visualization/hyperbolic.py [hunk 4]
```diff
@@ -120,6 +87,7 @@ def draw(self, ax, **kwargs):
                 raise ValueError("Points do not have dimension 2.")
 
     def plot(self, points, ax=None, point_type=None, **point_draw_kwargs):
```
Comment (luisfpereira) [inline geomstats/visualization/hyperbolic.py:89]: `point_type` should not be in signature, but at init.

File: geomstats/visualization/hyperbolic.py [hunk 5]
```diff
@@ -130,25 +98,34 @@ def plot(self, points, ax=None, point_type=None, **point_draw_kwargs):
...
             self.add_points(points)
+        self._space = PoincareHalfSpace(dim=2)
+        self._metric = self._space.metric
+        self._belongs = self._space.belongs
+
+        self._ax_scale = 1.2
+        self._dim = 2
+        self._convert_points = self._convert_to_half_plane_coordinates
 
     def add_points(self, points):
```
Comment (luisfpereira) [inline geomstats/visualization/hyperbolic.py:118]: to be removed.

File: geomstats/visualization/hyperbolic.py [hunk 6]
```diff
@@ -179,15 +148,18 @@ def convert_to_half_plane_coordinates(points):
         return half_plane_coords
 
     def draw(self, ax, **kwargs):
```
Comment (luisfpereira) [inline geomstats/visualization/hyperbolic.py:150]: as above.

File: geomstats/visualization/hyperbolic.py [hunk 7]
```diff
@@ -179,15 +148,18 @@ def convert_to_half_plane_coordinates(points):
...
+        """ "Plot points for Poincare Half Plane"""
         points_x = [gs.to_numpy(point[0]) for point in self.points]
         points_y = [gs.to_numpy(point[1]) for point in self.points]
         ax.scatter(points_x, points_y, **kwargs)
 
     def plot(self, points, ax=None, point_type=None, **point_draw_kwargs):
```
Comment (luisfpereira) [inline geomstats/visualization/hyperbolic.py:156]: as above.

**Review by luisfpereira — CHANGES_REQUESTED** [review]
@alicemo1, several changes need to be done to make it consistent with `Plotter` class.

**Pull request closed without merging**